addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -65,11 +65,34 @@ class DataEgressService extends Service {
     this._scanner = () => dbService.helper.scanner().table(table);
   }
 
+  async getEgressStoreInfo(environmentId) {
+    const workspaceId = environmentId;
+    let egressStoreScanResult = [];
+
+    try {
+      egressStoreScanResult = await this._scanner()
+        .limit(1000)
+        .scan()
+        .then(egressStores => {
+          return egressStores.filter(store => store.workspaceId === workspaceId);
+        });
```

Comment:
I know we have done this before but I didn't notice before think this is something that won't scale with a lot of workspaces. My suggestion would be to use the hashkey here (I am guessing its workspaceId, right?)

Reply:
Yes, make sense, I will push another fix PR for this.